### PR TITLE
Kevin/mark scan fix flaky tests

### PR DESCRIPTION
**Pull request by kshen0**
## Overview

Fixes mark-scan flaky tests

* Converts Observables into child state machines
* Refactors mark-scan state machine to use `Clock` and `SimulatedClock`
* Removes `expectStatusTransitionTo`, designed to allow assertions on nonblocking states

## Demo Video or Screenshot
n/a

## Testing Plan
Existing tests

### CODE REVIEW

File: apps/mark-scan/backend/src/custom-paper-handler/state_machine.ts [hunk 1]
```diff
@@ -907,7 +923,7 @@ export function buildMachine(
                 PAPER_JAM: 'jammed',
               },
               after: {
-                [DELAY_BEFORE_DECLARING_REAR_JAM_MS]: 'jammed',
+                [delays.DELAY_BEFORE_DECLARING_REAR_JAM_MS]: 'jammed',
```
Comment (jonahkagan) [inline apps/mark-scan/backend/src/custom-paper-handler/state_machine.ts:934]: These could probably also just reference the machine's configured delay with that key now:
```ts
after: {
  DELAY_BEFORE_DECLARING_REAR_JAM_MS: 'jammed'
}
```
Not really any functional difference, so take it or leave it

File: apps/mark-scan/backend/src/custom-paper-handler/state_machine.test.ts [hunk 2]
```diff
@@ -228,6 +197,12 @@ afterEach(async () => {
   jest.resetAllMocks();
 });
 
+async function setMockStatusAndIncrementClock(status: MockPaperHandlerStatus) {
+  await sleep(0);
```
Comment (jonahkagan) [inline apps/mark-scan/backend/src/custom-paper-handler/state_machine.test.ts:201]: Maybe a comment explaining what this is for would be helpful

File: apps/mark-scan/backend/src/custom-paper-handler/state_machine.test.ts [hunk 3]
```diff
@@ -264,25 +239,31 @@ describe('eject_to_front', () => {
...
   it('transitions to loading_paper state when front sensors are triggered', async () => {
+    mockPollWorkerAuth(auth, electionGeneralDefinition);
+
+    expect(machine.getSimpleStatus()).toEqual('not_accepting_paper');
     machine.setAcceptingPaper(ACCEPTED_PAPER_TYPES);
     expect(machine.getSimpleStatus()).toEqual('accepting_paper');
 
-    driver.setMockStatus('paperInserted');
-    await expectStatusTransitionTo('loading_paper');
+    // await sleep(0);
```
Comment (jonahkagan) [inline apps/mark-scan/backend/src/custom-paper-handler/state_machine.test.ts:257]: Remove

File: apps/mark-scan/backend/src/custom-paper-handler/state_machine.test.ts [hunk 4]
```diff
@@ -292,11 +273,31 @@ describe('loading_paper', () => {
...
+    await waitForStatus('waiting_for_ballot_data');
+  });
+});
+
+it('logs when an auth error happens', async () => {
+  // Transition to a state that polls auth
+  mockOf(auth.getAuthStatus).mockRejectedValue(new Error('mock auth error'));
+  machine.setAcceptingPaper(ACCEPTED_PAPER_TYPES);
+
+  // mockOf(auth.getAuthStatus).mockRejectedValue(new Error('mock auth error'));
```
Comment (jonahkagan) [inline apps/mark-scan/backend/src/custom-paper-handler/state_machine.test.ts:293]: Remove?

File: apps/mark-scan/backend/src/custom-paper-handler/state_machine.test.ts [hunk 5]
```diff
@@ -292,11 +273,31 @@ describe('loading_paper', () => {
...
+  clock.increment(delays.DELAY_PAPER_HANDLER_STATUS_POLLING_INTERVAL_MS);
+
+  // Allow event loop to resolve `logger.log` promise before making assertions
+  await Promise.resolve();
+
+  expect(logger.log).toHaveBeenCalledWith(LogEventId.UnknownError, 'system', {
+    error: 'mock auth error',
   });
```
Comment (jonahkagan) [inline apps/mark-scan/backend/src/custom-paper-handler/state_machine.test.ts:301]: Could just use `backendWaitFor` to be a bit more resilient

**Review by jonahkagan — APPROVED** [review]
Thank you for taking this on! I know it was a bit of a slog. Heroic effort 💪 🏆

File: apps/mark-scan/backend/src/custom-paper-handler/state_machine.ts (hunk 1 above)
Comment (kshen0) [reply apps/mark-scan/backend/src/custom-paper-handler/state_machine.ts:934]: Mostly copied this construct over for consistency with VxScan. I don't think we need the interface but it's nice to have all the delays more obviously located in a single place (whereas before they were in a random constants file).

**Pull request merged**